### ISSUE TITLE
Only run TPC-H scale_factor = 10 when memory >= 64 GB

### ISSUE THREAD

**Issue by alistaire47** [voltrondata-labs/benchmarks#141]
TPC-H query 21 at scale factor 10 has been regularly failing on machines with less than 64 GB of memory, e.g. https://buildkite.com/apache-arrow/arrow-bci-benchmark-on-arm64-t4g-linux-compute/builds/2829#0188b088-42b8-4218-ae4a-93dd0058ce8c As a solution, remove scale_factor=10 permutations in `get_valid_cases()` when there is insufficient memory.

https://github.com/voltrondata-labs/benchmarks/blob/606f4fc986a320dc5c7c5ebefab2540fae578edb/benchmarks/tpch_benchmark.py#L6-L12

**Comment by jgehrcke** [comment]
> TPC-H query 21 at scale factor 10 has been regularly failing on machines with less than 64 GB of memory, e.g. https://buildkite.com/apache-arrow/arrow-bci-benchmark-on-arm64-t4g-linux-compute/builds/2829#0188b088-42b8-4218-ae4a-93dd0058ce8c

The issue tracking that is https://github.com/voltrondata-labs/arrow-benchmarks-ci/issues/64 right?

**Comment by alistaire47** [comment]
Yep, thanks for the link! This is the repo that will need the PR though.
